sources/web/datalab/polymer/components/item-list/item-list.ts

```diff
@@ -246,6 +254,34 @@ class ItemListElement extends Polymer.Element {
     });
   }
 
+  _rowsChanged() {
+    // Reset the filter if the items change
```

Comment:
We are auto-refreshing the file list, so if it happens to change while we are using filtering, this code will clear our filtering expression. That seems like it would be annoying. Maybe we can have a flag so we can tell whether this change to the file list is happening due to user action as opposed to auto-refresh, and only clear the filter when it is based on user action.

Reply:
We only set `rows` if it will actually change to avoid changing current selection. This shouldn't have an effect.

Reply:
Yes, I am referring to the situation in which rows actually changes. The scenario is, I am looking at a shared folder with 300 files in it, and I have entered a filter string to get it down to 10 files, then someone else adds a new file to that folder, and my client auto-refreshes. In that case, although I have not done anything, I will suddenly lose my filter string and be back to looking at 301 files.

Reply:
Good point. I'll move resetting the filter to be done from outside the component, and have the file browser component reset it only when changing directory.